UnityProject/Assets/Scripts/Managers/ComponentManager.cs

```diff
@@ -7,7 +7,28 @@
 public class ComponentManager : SingletonManager<ComponentManager>
 {
 	public static Dictionary<GameObject, UniversalObjectPhysics> ObjectToPhysics = new Dictionary<GameObject, UniversalObjectPhysics>();
-	//TODO in future maybe reference a Component that is on every prefab that handles all Components on it
+	public static Dictionary<GameObject, CommonComponents> ObjectToCommonComponent = new Dictionary<GameObject, CommonComponents>();
+
+
+
+	public static bool TryGetCommonComponent(GameObject gameObject, out CommonComponents CommonComponents)
```

Comment:
what is this for?

Reply:
basically a way to get around get component 
for threads and 
for performance (dictionary vs whatever unity is doing)